src/Native/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -249,16 +249,18 @@ bool CoffNativeCodeManager::IsFunclet(MethodInfo * pMethInfo)
     return (unwindBlockFlags & UBF_FUNC_KIND_MASK) != UBF_FUNC_KIND_ROOT;
 }
 
-PTR_VOID CoffNativeCodeManager::GetFramePointer(MethodInfo *   pMethodInfo,
+PTR_VOID CoffNativeCodeManager::GetFramePointer(MethodInfo *   pMethInfo,
                                          REGDISPLAY *   pRegisterSet)
 {
-    // If the method has EHinfo then it is guaranteed to have a frame pointer.
-    CoffNativeMethodInfo * pNativeMethodInfo = (CoffNativeMethodInfo *)pMethodInfo;
+    CoffNativeMethodInfo * pMethodInfo = (CoffNativeMethodInfo *)pMethInfo;
+
+    size_t unwindDataBlobSize;
+    PTR_VOID pUnwindDataBlob = GetUnwindDataBlob(m_moduleBase, pMethodInfo->runtimeFunction, &unwindDataBlobSize);
 
-    // Get to unwind info
-    PTR_UNWIND_INFO pUnwindInfo(dac_cast<PTR_UNWIND_INFO>(m_moduleBase + pNativeMethodInfo->runtimeFunction->UnwindInfoAddress));
+    uint8_t unwindBlockFlags = *(dac_cast<DPTR(uint8_t)>(pUnwindDataBlob) + unwindDataBlobSize);
```

Comment:
This is fixing bug in handling of nested exceptions that I have run into along the way. 

@janvorli Could you please take a look at this change? I think that the ELF code manager that you are working on should use the same logic. The frame pointer returned from here is just used to drive EH, so it just needs to be returned for anything that contains EH.